### ISSUE TITLE
Not compatible with Rakudo 2023.04 and older

### ISSUE THREAD

**Issue by 2colours** [finanalyst/raku-pod-render#25]
With https://github.com/finanalyst/raku-pod-render/commit/466dad518d0d5454b5526eb14880a71e36d359d9, the code references `Pod::Block::Input` which is a type only introduced with https://github.com/rakudo/rakudo/commit/6f482ca006bc837c48bb99f6259cca94ec7bb6c4, hence the dist installation will fail with older versions of Rakudo.

Actually, there should be a strategy to prevent situations like this. I don't know whether Rakudo shouldn't create situations like this, the dist should be more conservative, or the dist should simply say the appropriate compatibility level.

**Comment by finanalyst** [comment]
@2colours Does `Raku::Pod::Render` actually fail on older versions of Rakudo? Liz told me that Roast contains tests of PBOutput and PBInput, but they were always mapped onto PBCode.  

No one uses PBInput or Output, so I can disable them for `Raku::Pod::Render`.

**Comment by 2colours** [comment]
@finanalyst yes, it does. Somebody reported this on IRC with 2023.04 and I can confirm. The type constraint used is invalid.

**Comment by finanalyst** [comment]
@2colors I've a lot of problems to work on at present and not much time.

I'd be happy to revert the change. How is the error being seen? How can I
test whether a workaround doesn't cause a similar problem?

Richard

**Comment by 2colours** [comment]
The direct error is that the signature of `handle` candidates refer types that didn't really exist a couple of months ago. The size and complexity of the commit doesn't really help me estimate the actual influence.

As I said, the error is easy to reproduce with rakudo v2023.04. I just set it up with rakubrew and then tried to install the dist with zef.

**Comment by finanalyst** [comment]
@2colours That explanation really helps. I added two new handles to deal with the new types. I had not expected there to be an error. 

I'll try to find a workaround.

**Comment by finanalyst** [comment]
@2colours I think the failure may be arising because the input/output is tested. I have made the tests compiler dependent, so compiler version 2023.04 or earlier does not render or test this POD.

I do not think `=output` or `=input` is used at all in the documentation suite, so an error should not arise.
